react/features/base/tracks/functions.js

```diff
@@ -10,6 +10,52 @@ import {
 
 import logger from './logger';
 
+/**
+ * Creates a local video track for presenter. The constraints are computed based
+ * on the height of the desktop that is being shared.
+ *
+ * @param {Object} options - The options with which the local presenter track
+ * is to be created.
+ * @param {string|null} [options.cameraDeviceId] - Camera device id or
+ * {@code undefined} to use app's settings.
+ * @param {number} desktopHeight - The height of the desktop that is being
+ * shared.
+ * @returns {Promise<JitsiLocalTrack>}
+ */
+export async function createLocalPresenterTrack(options, desktopHeight) {
+    const { cameraDeviceId } = options;
+
+    // compute the constraints of the camera track based on the resolution
+    // of the desktop screen that is being shared.
+    const cameraHeights = [ 180, 270, 360, 540, 720 ];
+    const proportion = 4;
+    const result = cameraHeights.find(
+            height => (desktopHeight / proportion) < height);
+    const constraints = {
+        video: {
+            aspectRatio: 4 / 3,
+            height: {
+                exact: result
+            }
+        }
+    };
+
+    try {
+        const [ videoTrack ] = await JitsiMeetJS.createLocalTracks(
+            {
+                cameraDeviceId,
+                constraints,
+                devices: [ 'video' ]
+            });
+
+        videoTrack.type = MEDIA_TYPE.PRESENTER;
+
+        return videoTrack;
+    } catch (err) {
+        return err;
```

Comment:
remove the try catch instead of returning the error. As it stands, it would resolve the promise succcesfully, but with an error object. We want it to be rejected instead. Not having the try will make an eventual exception reject the promise.